From 5f2d58d8ba5fa346290ffca087b598c14cf08a90 Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Thu, 2 Jan 2025 10:05:39 +0800
Subject: [PATCH 87/92] mtk: mt76: mt7996: qos: fill the User Priority in
 skb->priority for rx packet

Set the UP in skb->priority to allow MSCS DSCP Learning to observe user
priority at the upper layer, mac80211 or netfilter pre-routing hook
point.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>

---
 mac80211.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mac80211.c b/mac80211.c
index 17d07e15..7100122b 100644
--- a/mac80211.c
+++ b/mac80211.c
@@ -1203,6 +1203,8 @@ mt76_rx_convert(struct mt76_dev *dev, struct sk_buff *skb,
 	mstat = *((struct mt76_rx_status *)skb->cb);
 	memset(status, 0, sizeof(*status));
 
+	skb->priority = mstat.qos_ctl & IEEE80211_QOS_CTL_TID_MASK;
+
 	status->flag = mstat.flag;
 	status->freq = mstat.freq;
 	status->enc_flags = mstat.enc_flags;
-- 
2.45.2

